src/wp-login.php

```diff
@@ -1267,7 +1267,10 @@ function wp_login_viewport_meta() {
 				$errors->add( 'enter_recovery_mode', __( 'Recovery Mode Initialized. Please log in to continue.' ), 'message' );
 			} elseif ( isset( $_GET['redirect_to'] ) && false !== strpos( $_GET['redirect_to'], 'wp-admin/authorize-application.php' ) ) {
 				$query_component = wp_parse_url( $_GET['redirect_to'], PHP_URL_QUERY );
-				parse_str( $query_component, $query );
+				$query           = array();
+				if ( $query_component ) {
```

Comment:
What if the query component is `'0'`? https://3v4l.org/JaO38 then the `parse_str()` would not run. It's an odd query, yes, but possible in the wild.

```suggestion
				if ( is_string( $query_component ) && '' !== $query_component ) {
```

Reply:
But that said, there's no value `'0'` to the process. And if it's `'0'` instead of `key=value`, there's no value in doing `parse_str()`. It would flow through to the `else` message whether the above change happens or not.